service/model/project_membership.go

```diff
@@ -115,3 +131,64 @@ func (i ProjectInvitationStatus) Validate() error {
 
 	return errProjectInvitationStatusInvalid
 }
+
+func (r ProjectRole) IsRoleAtLeast(role ProjectRole) bool {
+	return projectRolePriority[r] <= projectRolePriority[role]
+}
+
+type ProjectMember struct {
+	User        User
+	ProjectID   uuid.UUID
```

Comment:
(@bolshakoff in the end I decided to handle project member creation in Go.)

I went back and forth on this, but I think it's good to include `User` as part of the `ProjectMember` struct. In most cases, I will need at least the user’s name and email every time I work with a project member. And knowing the user's role might be useful, as it can override the project role in the case of an admin user role.